internal/resources/cluster/data_source_cluster.go

```diff
@@ -131,11 +131,17 @@ func dataSourceTMCClusterRead(ctx context.Context, d *schema.ResourceData, m int
 
 	switch {
 	case resp.Cluster.Spec.TkgAws != nil:
-		resp.Cluster.Spec.TkgAws.Topology.NodePools = clusterSpec.TkgAws.Topology.NodePools
+		if clusterSpec != nil && clusterSpec.TkgAws != nil && clusterSpec.TkgAws.Topology != nil {
```

Comment:
Here, resp should also be validated for nil. 
custerSpec nil check can also be done at constructSpec also. Not sure just check